worldedit-core/src/main/java/com/sk89q/worldedit/command/GenerationCommands.java

```diff
@@ -587,11 +594,31 @@ public void image(
         if (!url.getHost().equalsIgnoreCase("i.imgur.com")) {
             throw new IOException("Only i.imgur.com links are allowed!");
         }
-        BufferedImage image = MainUtil.readImage(url);
-        if (dimensions != null) {
-            image = ImageUtil.getScaledInstance(image, dimensions.getBlockX(), dimensions.getBlockZ(),
-                    RenderingHints.VALUE_INTERPOLATION_BILINEAR, false
-            );
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        Future<BufferedImage> future = executor.submit(() -> {
+            BufferedImage image = MainUtil.readImage(url);
+            if (dimensions != null) {
+                checkCommandArgument(
+                        (long) dimensions.getX() * dimensions.getZ() <= Settings.settings().WEB.MAX_IMAGE_SIZE,
+                        Caption.of("fawe.error.image-dimensions", Settings.settings().WEB.MAX_IMAGE_SIZE)
+                );
```

Comment:
This check could be done before starting a new thread, however it doesn't prevent someone from loading a large image without a dimension argument.

Reply:
The image itself is checked in the MainUtil method

Reply:
I see, missed that.